lib/players/computer.rb

```diff
@@ -0,0 +1,124 @@
+class Computer < Player
+  attr_accessor :opp_token, :valid_moves
+
+  def move(board)
+    self.valid_moves = ["1", "2", "3", "4", "5", "6", "7", "8", "9"]
+    if self.token == "X"
+      self.opp_token = "O"
+    else
+      self.opp_token = "X"
+    end
+    if self.offensive(board)
+      self.offensive(board)
+    elsif self.defensive(board)
+      self.defensive(board)
+    else
+      self.valid_moves.sample
+    end
+  end
+
+  def offensive(board)
```

Comment:
It would be a good exercise to look for ways to generalize your logic in `offensive` and `defensive`.